src/common/ceph_context.h

```diff
@@ -261,6 +262,15 @@ class CephContext {
   void notify_pre_fork();
   void notify_post_fork();
 
+  void set_mon_addrs(const std::list<entity_addrvec_t>& in) {
+    // protect updates to list, to avoid racy updates and gets?
+    _mon_addrs = in;
+  }
```

Comment:
```suggestion
  void set_mon_addrs(const std::vector<entity_addrvec_t>& in) {
    // protect updates to list, to avoid racy updates and gets?
    auto ptr = std::make_shared<std::vector<entity_addrvec_t>(in);
    atomic_store_explicit(&_mon_addrs, std::move(ptr), std::memory_order_relaxed);
  }
```